### ISSUE TITLE
[feature] instrument add-on discovery.

### ISSUE THREAD

**Issue by gabrielcsapo** [ember-cli/ember-cli#9373]
Currently the output from the initial build successful is not indicative of the time it took from starting the build to this point.

<img width="609" alt="Screen Shot 2020-10-23 at 8 32 18 PM" src="https://user-images.githubusercontent.com/1854811/97067081-d93a6200-156e-11eb-92e0-00dcc571913c.png">

It would be great to add instrumentation to the addon discovery to be able to get a more accurate representation of the total time it took to build. 

---

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 3.17.0
node: 14.2.0
v8: 8.1.307.31-node.33
uv: 1.37.0
zlib: 1.2.11
brotli: 1.0.7
ares: 1.16.0
modules: 83
nghttp2: 1.40.0
napi: 6
llhttp: 2.0.4
openssl: 1.1.1g
cldr: 36.1
icu: 66.1
tz: 2019c
unicode: 13.0
os: darwin x64
6.14.4
1.22.4
```

**Comment by rwjblue** [comment]
Yes, we should totally do this. Basically we should instrument (via Heimdall nodes) both the discovery and the instantiation (separate nodes so we can disambiguate).

We should **probably** also consider instrumenting some other hooks in the addon base class as well, I'm thinking `included` and `treeFor` (which will cover all of the `treeFor*` methods) at least.
